src/accelerate/accelerator.py

```diff
@@ -74,10 +84,18 @@ def __init__(
         split_batches: bool = False,
         fp16: bool = None,
         cpu: bool = False,
+        deepspeed_plugin: DeepSpeedPlugin = None,
         rng_types: Optional[List[Union[str, RNGType]]] = None,
         kwargs_handlers: Optional[List[KwargsHandler]] = None,
     ):
-        self.state = AcceleratorState(fp16=fp16, cpu=cpu, _from_accelerator=True)
+        if deepspeed_plugin is None:
+            deepspeed_plugin = DeepSpeedPlugin()  # init from env variables
+        else:
+            assert isinstance(
+                deepspeed_plugin, DeepSpeedPlugin
+            ), "`deepspeed_plugin` must be a DeepSpeedPlugin object."
```

Comment:
Do we really need a `DeepSpeedPlugin` to be initialized if deepspeed isn't available?